test/framework/rootpod_executor.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"fmt"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/mock/go/context"
+	"github.com/sirupsen/logrus"
+	"io/ioutil"
+	corev1 "k8s.io/api/core/v1"
+	"time"
+)
+
+// RootPodExecutor enables the execution of command on the operating system of a node.
+// The executor deploys a pod with root privileged on a specified node.
+// This pod is then used to execute commands on the host operating system.
+type RootPodExecutor struct {
+	logger    *logrus.Logger
+	client    kubernetes.Interface
+	nodeName  *string
+	namespace string
+
+	Pod *corev1.Pod
+}
+
+// NewRootPodExecutor creates a new root pod executor to run commands on a node.
+func NewRootPodExecutor(logger *logrus.Logger, c kubernetes.Interface, nodeName *string, namespace string) (*RootPodExecutor, error) {
+	return &RootPodExecutor{
+		logger:    logger,
+		client:    c,
+		nodeName:  nodeName,
+		namespace: namespace,
+	}, nil
+}
+
+// Deploy deploys a root pod on the specified node and waits until it is running
+func (e *RootPodExecutor) Deploy(ctx context.Context) error {
+	rootPod, err := DeployRootPod(ctx, e.client.Client(), e.namespace, e.nodeName)
+	if err != nil {
+		return err
+	}
+	if err := WaitUntilPodIsRunning(ctx, e.logger, rootPod.Name, rootPod.Namespace, e.client); err != nil {
+		return err
+	}
+
+	e.Pod = rootPod
+	return nil
+}
+
+// Clean delete the deployed pod
+func (e *RootPodExecutor) Clean(ctx context.Context) error {
+	if e.Pod == nil {
+		return nil
+	}
+
+	return DeleteAndWaitForResource(ctx, e.client, e.Pod, 2*time.Minute)
+}
+
+// RunCommand executes a command on the node the root pod is running
+func (e *RootPodExecutor) RunCommand(ctx context.Context, command string) ([]byte, error) {
+	if e.Pod == nil {
+		if err := e.Deploy(ctx); err != nil {
+			return nil, err
+		}
+	}
+
+	command = fmt.Sprintf("chroot /hostroot %s", command)
+	reader, err := kubernetes.NewPodExecutor(e.client.RESTConfig()).Execute(ctx, e.Pod.Namespace, e.Pod.Name, e.Pod.Spec.Containers[0].Name, command)
```

Comment:
This seems kind of fishy, to hand the `RESTConfig` of a client to the executor, which then creates a new client.
If I see it correctly `kubernetes.NewPodExecutor()` is only used in `/test`, so can you move it to `test/framework` and refactor it a bit to reuse an existing client, please?

Reply:
PTAL